pkg/sentry/mm/vma.go

```diff
@@ -214,7 +216,9 @@ func (mm *MemoryManager) findLowestAvailableLocked(length, alignment uint64, bou
 
 // Preconditions: mm.mappingMu must be locked.
```

Comment:
Note that you can also fix the pgalloc issue with the same root cause, code here:
https://cs.opensource.google/gvisor/gvisor/+/master:pkg/sentry/pgalloc/pgalloc.go;drc=238e80fe38670faed5418444ce9fb2eb84e6d5d7;bpv=0;bpt=1;l=456

Right now it is traversing segments, but it could easily be adapted to traverse gaps and use the NextLargeEnoughGap function that you've built.

Reply:
Probably can done in another pull request?